### DISCUSSION TITLE
Can I use Android App to read files?

### DISCUSSION THREAD

**Discussion by tigerjenny** [microsoft/WSA#512] [General]
My Android App has a function that requires reading files, such as Json files, and I placed the files in the Windows folder.
I currently try to use ACTION_OPEN_DOCUMENT, but it will close the APP directly after execution.
I'm not sure if the Roadmap showing unavailable File transfer means that reading files is not supported.

This is my first time using Android APP on Windows system. Please help me clear up my confusion, thank you.
Or if you have any relevant examples that you can refer to, please tell me and I will be very grateful to you.

**Comment by OurCodeBase** [comment]
File Sharing Now Available ! in #321

**Reply by tigerjenny** [reply]
private void getFile() {
        fileIntent = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
            if (null != result.getData()) {
                Uri uri = result.getData().getData();
                File file = new File(uri.getPath());
                FileInputStream inputStream;

                try {
                    inputStream = new FileInputStream(file);
                    byte temp[] = new byte[1024];
                    StringBuilder sb = new StringBuilder("");
                    int len = 0;
                    while ((len = inputStream.read(temp)) > 0){
                        sb.append(new String(temp, 0, len));
                    }
                    Toast.makeText(this, "readSaveFile: \n" + sb.toString(), Toast.LENGTH_SHORT).show();
                    inputStream.close();

                } catch (Exception e) {
                    e.printStackTrace();
                    Toast.makeText(this, "readSaveFile: \n" + e.toString(), Toast.LENGTH_SHORT).show();
                }
            }
        });

        try {
            Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT);
            intent.addCategory(Intent.CATEGORY_OPENABLE);
            intent.setType("text/plain");
            fileIntent.launch(intent);
        } catch (Exception e) {
            Toast.makeText(this, "intent: \n" + e.toString(), Toast.LENGTH_SHORT).show();
        }
    }


This is my function, but when it open the file explorer the app was closed.
Is there anything wrong?
